sdk/clientcore/annotation-processor/src/main/java/io/clientcore/annotation/processor/AnnotationProcessor.java

```diff
@@ -207,58 +208,73 @@ private HttpRequestContext createHttpRequestContext(ExecutableElement requestMet
         return method;
     }
 
-    private void setReturnTypeFormMethod(HttpRequestContext method, ExecutableElement requestMethod,
+    private void setReturnTypeForMethod(HttpRequestContext method, ExecutableElement requestMethod,
         TemplateInput templateInput) {
-        // Get the return type from the method
         TypeMirror returnType = requestMethod.getReturnType();
+        String returnTypeString = returnType.toString();
```

Comment:
I get nervous when I see strings being used - you probably want to hang on to `returnType` and use more type-safe ways, e.g. `returnType.getKind() == TypeKind.VOID`